amplify/backend/api/colonycdapp/schema.graphql

```diff
@@ -8,492 +8,967 @@ input AMPLIFY {
+
+Definitions:
+* Inactive: balance of a token that is in a users wallet but can't be used by the Colony Network (e.g. for governance)
+* Locked: balance of a token that is staked (e.g. in motions)
+* Active: balance that is approved for the Colony Network to use (e.g. for governance)
+* Pending: [needs input]
+"""
 type GetUserTokenBalanceReturn {
-  balance: String # total balance, sum of inactive, locked and active
+  """The total token balance, including inactive, locked, and active balances."""
+  balance: String
+  """The inactive portion of the user's token balance."""
   inactiveBalance: String
+  """The locked portion of the user's token balance."""
   lockedBalance: String
+  """The active portion of the user's token balance."""
   activeBalance: String
+  """The pending portion of the user's token balance."""
   pendingBalance: String
 }
 
-# Definitions:
-# Member = User watching a Colony, with or without reputation
-# Contributor = User watching a Colony WITH reputation
-# Watcher = User watching a Colony WITHOUT reputation
+"""
+A return type representing the members of a Colony.
+
+Definitions:
+* Member = User watching a Colony, with or without reputation
+* Contributor = User watching a Colony WITH reputation
+* Watcher = User watching a Colony WITHOUT reputation
+"""
 type MembersForColonyReturn {
+  """User watching a Colony WITH reputation"""
   contributors: [Contributor!]
+  """User watching a Colony WITHOUT reputation"""
   watchers: [Watcher!]
 }
 
+"""
+Variants of different token types a Colony can use.
+As Colonies can use multiple tokens and even own tokens (BYOT), we need to differentiate.
+"""
 enum TokenType {
+  """A (ERC20-compatible) token that was deployed with Colony. It has a few more features, like minting through the Colony itself"""
   COLONY
+  """An ERC20-compatible token"""
   ERC20
+  """The native token of the Chain used (e.g. ETH on mainnet or xDAI on Gnosis-Chain)"""
   CHAIN_NATIVE
 }
 
+
+"""
+Variants of supported Ethereum networks.
+"""
 enum Network {
+  """Local development network using Ganache."""
   GANACHE
+  """Ethereum Mainnet."""
   MAINNET
+  """Ethereum Goerli test network."""
   GOERLI
+  """Gnosis Chain network."""
   GNOSIS
+  """Fork of Gnosis Chain for QA purposes."""
   GNOSISFORK
 }
 
+"""
+Variants of available domain colors as used in the dApp.
+"""
 enum DomainColor {
+  """A light pink color."""
   LIGHT_PINK
+  """A pink color."""
   PINK
+  """A black color."""
   BLACK
+  """An emerald green color."""
   EMERALD_GREEN
+  """A blue color."""
   BLUE
+  """A yellow color."""
   YELLOW
+  """A red color."""
   RED
+  """A green color."""
   GREEN
+  """A pale indigo color."""
   PERIWINKLE
+  """A gold color."""
   GOLD
+  """An aqua color."""
   AQUA
+  """A blue-grey(ish) color."""
   BLUE_GREY
+  """A purple color."""
   PURPLE
+  """An orange color."""
   ORANGE
+  """A magenta color."""
   MAGENTA
+  """A purple-grey(ish) color."""
   PURPLE_GREY
```

Comment:
Is the convention to document _every_ piece of the schema, even if it's entirely self-explanatory?

Reply:
I would recommend doing it for every piece, for two reasons: first, it removes the mental overhead for the developer of thinking about _whether_ a model needs annotating or not. Second of all, it levels the bias that a developer might have as to whether a piece of code is self-explanatory or not (people who have worked on it or named the variables might think that it's far easier to understand than someone who's new to the code base - or even a dev who isn't familiar with any of the concepts at all). These colors are a very clear case of which there aren't many.